src/contexts/ethereum-provider.context.tsx

```diff
@@ -0,0 +1,120 @@
+import { createContext, FC, useContext, useEffect, useState } from "react";
+import { Web3Provider } from "@ethersproject/providers";
+import WalletConnectProvider from "@walletconnect/ethereum-provider";
+import { useNavigate } from "react-router-dom";
+
+import { EthereumEvent, WalletName } from "src/domain";
+import { AsyncTask } from "src/utils/types";
+import { ethereumAccountsParser } from "src/adapters/parsers";
+import { getConnectedAccounts } from "src/adapters/ethereum";
+import { useEnvContext } from "./env.context";
+import routes from "src/routes";
+
+interface EthereumProviderContextData {
+  provider?: Web3Provider;
+  account: AsyncTask<string, string>;
+  connectProvider?: (walletName: WalletName) => Promise<void>;
+}
+
+const EthereumProviderContext = createContext<EthereumProviderContextData>({
+  provider: undefined,
+  account: { status: "pending" },
```

Comment:
Why are we explicit with `provider: undefined` and not with `connectProvider: undefined`?

Reply:
Good catch! Fixed